packages/core/__tests__/knapsack-pro-env.config.spec.ts

```diff
@@ -173,96 +265,114 @@ describe('KnapsackProEnvConfig', () => {
+    describe('when KNAPSACK_PRO_BRANCH is defined on the environment AND the branch is defined on a supported CI environment AND they are the same value', () => {
+      it('does not log info', () => {
+        process.env.KNAPSACK_PRO_BRANCH = 'aaa111';
+        process.env.BUILDKITE = 'true';
+        process.env.BUILDKITE_BRANCH = 'aaa111';
 
-    beforeEach(() => {
-      // eslint-disable-next-line dot-notation
-      KnapsackProEnvConfig['$knapsackProLogger'] = {
-        info(message) {
-          logs.push(message);
-        },
-      } as KnapsackProLogger;
+        expect(logs).toEqual([]);
+      });
     });
+  });
 
+  describe('.fixedQueueSplit', () => {
     afterEach(() => {
       // eslint-disable-next-line dot-notation
       KnapsackProEnvConfig['$fixedQueueSplit'] = undefined;
-      logs = [];
     });
 
     describe('when ENV exists', () => {
       describe('when KNAPSACK_PRO_FIXED_QUEUE_SPLIT=false', () => {
-        const TESTS: [string, object][] = [
-          ['AppVeyor', { APPVEYOR: 'whatever' }],
-          ['Buildkite', { BUILDKITE: 'whatever' }],
-          ['CircleCI', { CIRCLECI: 'whatever' }],
-          ['Cirrus CI', { CIRRUS_CI: 'whatever' }],
-          ['Codefresh', { CF_BUILD_ID: 'whatever' }],
-          ['Codeship', { CI_NAME: 'codeship' }],
-          ['GitHub Actions', { GITHUB_ACTIONS: 'whatever' }],
-          ['GitLab CI', { GITLAB_CI: 'whatever' }],
-          ['Heroku CI', { HEROKU_TEST_RUN_ID: 'whatever' }],
-          ['Semaphore CI 1.0', { SEMAPHORE_BUILD_NUMBER: 'whatever' }],
+        const TESTS: [string, object, boolean][] = [
+          ['AppVeyor', { APPVEYOR: 'whatever' }, false],
+          ['Buildkite', { BUILDKITE: 'whatever' }, true],
+          ['CircleCI', { CIRCLECI: 'whatever' }, false],
+          ['Cirrus CI', { CIRRUS_CI: 'whatever' }, false],
+          ['Codefresh', { CF_BUILD_ID: 'whatever' }, false],
+          ['Codeship', { CI_NAME: 'codeship' }, true],
+          ['GitHub Actions', { GITHUB_ACTIONS: 'whatever' }, true],
+          ['GitLab CI', { GITLAB_CI: 'whatever' }, true],
+          ['Heroku CI', { HEROKU_TEST_RUN_ID: 'whatever' }, false],
+          ['Semaphore CI 1.0', { SEMAPHORE_BUILD_NUMBER: 'whatever' }, false],
           [
             'Semaphore CI 2.0',
             { SEMAPHORE: 'whatever', SEMAPHORE_WORKFLOW_ID: 'whatever' },
+            false,
           ],
-          ['Travis CI', { TRAVIS: 'whatever' }],
-          ['Unsupported CI', {}],
+          ['Travis CI', { TRAVIS: 'whatever' }, true],
+          ['Unsupported CI', {}, true],
         ];
-        TESTS.forEach(([ci, env]) => {
+        TESTS.forEach(([ci, env, ciEnv]) => {
           it(`on ${ci} it is false`, () => {
             process.env = { ...process.env, ...env };
             process.env.KNAPSACK_PRO_FIXED_QUEUE_SPLIT = 'false';
 
             expect(KnapsackProEnvConfig.fixedQueueSplit).toEqual(false);
-            expect(logs).toEqual([]);
+
+            if (ciEnv === false) {
```

Comment:
If you go with the [same trick as in Ruby](https://github.com/KnapsackPro/knapsack_pro-ruby/pull/254#discussion_r1611246482), please see if you can do the same here and in the next test.